firestore/app/src/main/java/com/google/example/firestore/DocSnippets.java

```diff
@@ -1335,4 +1335,92 @@ public void onComplete(@NonNull Task<AggregateQuerySnapshot> task) {
         });
         // [END count_aggregate_query]
     }
+
+    public void orQuery() {
+        CollectionReference collection = db.collection("cities");
+        // [START or_queries]
+        Query query = collection.where(Filter.and(
+            Filter.greaterThan("name", "L"),
+            Filter.or(
+                Filter.equalTo("capital", true),
+                Filter.greaterThanOrEqualTo("population", 1000000)
+            )
+        ));
+        // [END or_queries]
+    }
+
+    public void orQueryDisjunctions() {
+        CollectionReference collection = db.collection("cities");
+
+        // [START one_disjunction]
+        collection.whereEqualTo("a", 1);
+        // [END one_disjunction]
+
+        // [START two_disjunctions]
+        collection.where(Filter.or(
+            Filter.equalTo("a", 1),
+            Filter.equalTo("b", 2)
+        ));
+        // [END two_disjunctions]
+
+        // [START four_disjunctions]
+        collection.where(Filter.or(
+            Filter.and(
+                Filter.equalTo("a", 1),
+                Filter.equalTo("c", 3)
+            ),
+            Filter.and(
+                Filter.equalTo("a", 1),
+                Filter.equalTo("d", 4)
+            ),
+            Filter.and(
+                Filter.equalTo("b", 2),
+                Filter.equalTo("c", 3)
+            ),
+            Filter.and(
+                Filter.equalTo("b", 2),
+                Filter.equalTo("d", 4)
+            )
+        ));
+        // [END four_disjunctions]
+
+        // [START four_disjunctions_compact]
+        collection.where(Filter.and(
+            Filter.or(
+                Filter.equalTo("a", 1),
+                Filter.equalTo("b", 2)
+            ),
+            Filter.or(
+                Filter.equalTo("c", 3),
+                Filter.equalTo("d", 4)
+            )
+        ));
+        // [END four_disjunctions_compact]
+
+        // [START 20_disjunctions]
+        collection.where(Filter.or(
+            Filter.inArray("a", Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)),
+            Filter.inArray("b", Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10)),
+        ));
```

Comment:
Unfortunately there's currently a bug in the backend which prevents this query from being served -- Even though it is a valid query (it's valid to have one `IN` operator per disjunction).

Reply:
We can add this to the snippet registry and then exclude it from devsite for now.